community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/CostBasedPipeBuilderFactory.scala

```diff
@@ -35,15 +35,20 @@ object CostBasedPipeBuilderFactory {
              tokenResolver: SimpleTokenResolver = new SimpleTokenResolver(),
              plannerName: Option[CostBasedPlannerName],
              runtimeBuilder: RuntimeBuilder,
-             useErrorsOverWarnings: Boolean
+             useErrorsOverWarnings: Boolean,
+             idpMaxTableSize: Int,
+             idpIterationDuration: Long
     ) = {
 
     def createQueryGraphSolver(n: CostBasedPlannerName): QueryGraphSolver = n match {
       case IDPPlannerName =>
-        IDPQueryGraphSolver(monitors.newMonitor[IDPQueryGraphSolverMonitor]())
+        IDPQueryGraphSolver(monitors.newMonitor[IDPQueryGraphSolverMonitor](), solverConfig = DefaultIDPSolverConfig(
+          maxTableSize = idpMaxTableSize,
+          iterationDuration = idpIterationDuration
```

Comment:
I changed it to ConfigurableIDPSolverConfig, and removed any default parameters.